e2e/cypress/integration/multi_team_and_dm/close_current_dm_redirects_spec.js

```diff
@@ -0,0 +1,154 @@
+
+        // * Expect channel title and url to secondDMUser's username
+        expectActiveChannelToBe(secondDMUser.username, `/messages/@${secondDMUser.username}`);
+    });
+
+    it('MM-T453 Changing URL to root url when viewing a direct message should redirect to direct message', () => {
+        // # Login as test user
+        cy.apiLogin(testUser);
+
+        // # View 'off-topic' channel
+        cy.visit(offTopicChannelUrl);
+
+        // # Send a direct message to a first user
+        sendDirectMessageToUser(firstDMUser, 'hi');
+
+        // # Visit root url
+        cy.visit('/');
+
+        // * Expect channel title and url to firstDMUser's username
+        expectActiveChannelToBe(firstDMUser.username, `/messages/@${firstDMUser.username}`);
+    });
+});
+
+const expectActiveChannelToBe = (title, url) => {
+    // * Expect channel title to match title passed in argument
+    cy.get('#channelHeaderTitle').
+        should('be.visible').
+        and('contain.text', title);
+
+    // * Expect url to match url passed in argument
+    cy.url().should('contain', url);
+};
+
+const sendDirectMessageToUser = (user, message) => {
+    // # Open a new direct message with firstDMUser
+    cy.get('#addDirectChannel').click();
+
+    // # Type username
+    cy.get('#selectItems input').should('be.enabled').type(`@${user.username}`, {force: true});
+
+    // * Expect user count in the list to be 1
+    cy.get('#multiSelectList').
+        should('be.visible').
+        children().
+        should('have.length', 1);
+
+    // # Select first user in the list
+    cy.get('body').
+        type('{downArrow}').
+        type('{enter}');
+
+    // # Click on "Go" in the group message's dialog to begin the conversation
+    cy.get('#saveItems').click();
+
+    // * Expect the channel title to be the user's username
+    // In the channel header, it seems there is a space after the username, justifying the use of contains.text instead of have.text
+    cy.get('#channelHeaderTitle').should('be.visible').and('contain.text', user.username);
+
+    // # Type message and send it to the user
+    cy.get('#post_textbox').
+        type(message).
+        type('{enter}');
+};
+
+const closeDirectMessageViaXButton = (sender, recipient, team) => {
+    // # Find the username in the 'Direct Messages' list and trigger the 'x' button to appear (hover over the username)
+    cy.apiGetChannelsForUser(sender.id, team.id).then((response) => {
+        const channels = response.body;
+
+        // Get the name of the channel to build the CSS selector for that specific DM link in the sidebar
+        const channelDmWithFirstUser = channels.find((channel) =>
+            channel.type === 'D' && channel.name.includes(recipient.id),
+        );
+
+        // # Close the DM via 'x' button next to username in direct message list
+        cy.get(`#sidebarItem_${channelDmWithFirstUser.name} .btn-close`).
+            click({force: true});
```

Comment:
I have to use force here because the 'x' is hidden (via CSS), I tried `invoke('show').click()` as suggested in some github threads but had mixed result. This seems to work everytime.

Reply:
Thanks, looks good to me.